src/scent_map.h

```diff
@@ -0,0 +1,63 @@
+#ifndef SCENT_H
+#define SCENT_H
+
+#include "enums.h"
+#include "game_constants.h"
+#include "cursesdef.h"
+
+#include <array>
+
+class map;
+
+class scent_map
+{
+    protected:
+        template<typename T>
+        using scent_array = std::array<std::array<T, SEEY *MAPSIZE>, SEEX *MAPSIZE>;
+
+        scent_array<int> grscent;
+        int null_scent = 0;
```

Comment:
I see now. That's for assignments like `scent( pos ) = num`. Looks like a WTF to me, but appears to be widespread.